client/src/app/payroll/payroll-list/state/payroll-list.model.ts

```diff
@@ -0,0 +1,25 @@
+import { Department } from 'src/app/employee/employee-list/state/employee-list.model';
+
+export interface PayrollList {
+  id: string;
+  checkDate: Date;
```

Comment:
This is technically a `string` when it comes from the API since there is no `Date` type in JSON.

If you want to have a `Date` type internally, you can separate the representation of API types from APP types - kinda like separating Records (persistence) from Entities (domain).

What I often do is create a `Schema` type for what the API responds with and a different type for what the app works with.

Then I create functions to map between the `Schema` and app types.

When getting a response back from the API using the `HttpClient` I do something like:

```typescript
// mapping function

export function mapPayrollSchemaToPayroll(schema: PayrollSchema): Payroll {
  return {
    // map props ...
    checkDate: new Date(schema.checkDate),

    // map remaining props ...
  }
}

// service class

.pipe(
  mapListResponseToValues(mapPayrollSchemaToPayroll)
)
```

Reply:
Right now the client doesn't perform any logic on checkDate so i think changing it to a string for now is fine, if we end up needing to use the date for logic in the app, we can use the schema mapping concept then.